server/src/main/java/org/apache/uniffle/server/ShuffleTaskManager.java

```diff
@@ -409,14 +410,15 @@ public long getPartitionDataSize(String appId, int shuffleId, int partitionId) {
   public long requireBuffer(
       String appId, int shuffleId, List<Integer> partitionIds, int requireSize) {
     ShuffleTaskInfo shuffleTaskInfo = shuffleTaskInfos.get(appId);
-    if (shuffleTaskInfo != null) {
-      for (int partitionId : partitionIds) {
-        long partitionUsedDataSize = getPartitionDataSize(appId, shuffleId, partitionId);
-        if (shuffleBufferManager.limitHugePartition(
-            appId, shuffleId, partitionId, partitionUsedDataSize)) {
-          ShuffleServerMetrics.counterTotalRequireBufferFailedForHugePartition.inc();
-          return -1;
-        }
+    if (null == shuffleTaskInfo) {
+      return RequireBufferStatusCode.NO_REGISTER.statusCode();
+    }
+    for (int partitionId : partitionIds) {
```

Comment:
@zuston Could you take another look at this place?

Reply:
I think the change is right. But I will introduce new grpc status code like `BUFFER_LIMIT_FOR_HUGE_PARTITION` instead of `NO_BUFFER`